source/kraken/tests/fill_disruption_from_chaos_tests.cpp

```diff
@@ -0,0 +1,127 @@
+    powered by Canal TP (www.canaltp.fr).
+Help us simplify mobility and open public transport:
+    a non ending quest to the responsive locomotion way of traveling!
+
+LICENCE: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE fill_disruption_from_chaos_test
+#include <boost/test/unit_test.hpp>
+#include "routing/raptor_api.h"
+#include "ed/build_helper.h"
+#include "routing/tests/routing_api_test_data.h"
+#include "tests/utils_test.h"
+#include "routing/raptor.h"
+#include "type/data.h"
+#include "kraken/fill_disruption_from_chaos.h"
+#include "type/chaos.pb.h"
+#include "type/gtfs-realtime.pb.h"
+
+
+struct logger_initialized {
+    logger_initialized()   { init_logger(); }
+};
+BOOST_GLOBAL_FIXTURE( logger_initialized )
+
+namespace ntest = navitia::test;
+namespace bt = boost::posix_time;
+
+BOOST_AUTO_TEST_CASE(add_impact_on_stop_area) {
+    ed::builder b("20120614");
+    b.vj("A")("stop_area:stop1", 8*3600 +10*60, 8*3600 + 11 * 60)("stop_area:stop2", 8*3600 + 20 * 60 ,8*3600 + 21*60);
+    b.vj("A")("stop_area:stop1", 9*3600 +10*60, 9*3600 + 11 * 60)("stop_area:stop2", 9*3600 + 20 * 60 ,9*3600 + 21*60);
+    navitia::type::Data data;
+    b.generate_dummy_basis();
+    b.finish();
+    b.data->pt_data->index();
+    b.data->build_raptor();
+    b.data->build_uri();
+    b.data->meta->production_date = boost::gregorian::date_period(boost::gregorian::date(2012,6,14), boost::gregorian::days(7));
+
+
+    chaos::Disruption disruption;
+    disruption.set_id("test01");
+    auto* impact = disruption.add_impacts();
+    impact->set_id("impact_id");
+    auto* severity = impact->mutable_severity();
+    severity->set_id("severity");
+    severity->set_effect(transit_realtime::Alert_Effect_NO_SERVICE);
+    auto* object = impact->add_informed_entities();
+    object->set_pt_object_type(chaos::PtObject_Type_stop_area);
+    object->set_uri("stop_area:stop1");
+    auto* app_period = impact->add_application_periods();
+    app_period->set_start(ntest::to_posix_timestamp("20120614T123200"));
+    app_period->set_end(ntest::to_posix_timestamp("20120616T123200"));
+
+
+    navitia::add_disruption(disruption, *b.data->pt_data, *b.data->meta);
```

Comment:
some more checks (and comments) would be useful.

you can check that the disruption is there, that the adapted objects are correctly created, that the theoric one are not changed, ...